contracts/DINRegistry.sol

```diff
@@ -52,32 +46,48 @@ contract DINRegistry {
      */
     function DINRegistry(uint _genesis) {
         genesis = _genesis;
+        index = _genesis;
 
-        // Register the genesis DIN to Kiosk.
+        // Register the genesis DIN to Kiosk. This will represent a DIN product.
         records[genesis].owner = msg.sender;
     }
 
     /**
      * Register a new DIN.
-     * @param DIN The DIN to register.
-     * @param owner The account that will own the registered DIN.
+     * @param owner The account that will own the DIN.
      */
-    function registerDIN(uint DIN, address owner) only_registrar only_unregistered(DIN) {
-        records[DIN].owner = owner;
-        NewOwner(DIN, owner, msg.sender);
```

Comment:
DINRegistry has more control over what DINs get registered. This doesn't allow as much flexibility for registering random numbers in the future, but I'd rather have DINMarket conform to DINRegistry's rules than make its own rules for registering.